### PR TITLE
Implementation of egamma-like footprint removal in photon isolation

### PR DESCRIPTION
**Pull request by emanueledimarco**
Synchronised with e/gamma full framework version on Spring15 MC sample.

### CODE REVIEW

File: PhysicsTools/Heppy/python/analyzers/objects/PhotonAnalyzer.py [hunk 1]
```diff
@@ -270,7 +283,16 @@ def randomCone( self, event ):
...
-
+    def attachFootprintRemovedIsolation(self, gamma):
+        # cone deltar=0.3
+        gamma.ftprAbsIsoCharged03 = self.IsolationComputer.chargedAbsIso(gamma.physObj, 0.3, 0, 0.0);
+        gamma.ftprAbsIsoPho03     = self.IsolationComputer.photonAbsIsoRaw( gamma.physObj, 0.3, 0, 0.0);
+        gamma.ftprAbsIsoNHad03    = self.IsolationComputer.neutralHadAbsIsoRaw( gamma.physObj, 0.3, 0, 0.0);
+        if self.cfg_ana.gamma_isoCorr == "rhoArea":
+            gamma.ftprAbsIso03 = (max(gamma.ftprAbsIsoCharged03-gamma.rho*gamma.EffectiveArea03[0],0) + max(gamma.ftprAbsIsoPho03-gamma.rho*gamma.EffectiveArea03[1],0) + max(gamma.ftprAbsIsoNHad03 - gamma.rho*gamma.EffectiveArea03[2],0))
+        elif self.cfg_ana.gamma_isoCorr != 'raw':
+            raise RuntimeError, "Unsupported gamma_isoCorr name '" + str(self.cfg_ana.gamma_isoCorr) +  "'! For now only 'rhoArea', 'raw' are supported."
```
Comment (gpetruc) [inline PhysicsTools/Heppy/python/analyzers/objects/PhotonAnalyzer.py:295]: for the "raw" case, shouldn't you do gamma.ftprAbsIso03 = gamma.ftprAbsIsoCharged03 + gamma.ftprAbsIsoPho03 + gamma.ftprAbsIsoNHad03? now it seems to me you're not defining it

Comment (emanueledimarco) [inline PhysicsTools/Heppy/python/analyzers/objects/PhotonAnalyzer.py:295]: Yes, you are right. Done now, before the cases with corrections.

**Comment by gpetruc** [comment]
@mariadalfonso @aelwood @pandolf can some of you check if these photon-related changes are ok for you too?

**Comment by mariadalfonso** [comment]
@emanueledimarco can you update the #442 and #443 for conflict, then I will test.

**Comment by aelwood** [comment]
@gpetruc the changes look fine for us

**Comment by emanueledimarco** [comment]
@mariadalfonso done. Now going to update the CMGTools branch accordingly

**Pull request merged**